### PR TITLE
Fix BOS tokens, add arguments, support non-LLaMA models

### PR DESCRIPTION
**Pull request by mbalesni**
I found out I messed up pushing these changes, and they were hanging in a detached head while I thought they are merged into the `sita` branch. Pushing them now.

### CODE REVIEW

File: config/deepspeed.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "bf16": {
+    "enabled": false
```
Comment (tomekkorbak) [inline config/deepspeed.json:3]: Is there a reason we want to switch bf16 off?

**Review by tomekkorbak — APPROVED** [review]
looks good!

File: config/deepspeed.json (hunk 1 above)
Comment (tomekkorbak) [reply config/deepspeed.json:3]: nvm

**Comment by mbalesni** [comment]
Note: the Deepspeed config is WIP, don't mind the values there.

**Pull request merged**